rasa/nlu/server.py

```diff
@@ -31,324 +52,276 @@ def create_argument_parser():
+    _configure_logging(loglevel, logfile)
 
+    default_model_config = _load_default_config(default_config_path)
 
-class RasaNLU(object):
-    """Class representing Rasa NLU http server"""
+    @app.get("/")
+    async def hello(request):
+        """Main Rasa route to check if the server is online"""
+        return response.text("hello from Rasa NLU: " + rasa.__version__)
 
-    app = Klein()
-
-    def __init__(
-        self,
-        data_router,
-        loglevel="INFO",
-        logfile=None,
-        num_threads=1,
-        token=None,
-        cors_origins=None,
-        testing=False,
-        default_config_path=None,
-    ):
-
-        self._configure_logging(loglevel, logfile)
-
-        self.default_model_config = self._load_default_config(default_config_path)
-
-        self.data_router = data_router
-        self._testing = testing
-        self.cors_origins = cors_origins if cors_origins else ["*"]
-        self.access_token = token
-        reactor.suggestThreadPoolSize(num_threads * 5)
-
-    @staticmethod
-    def _load_default_config(path):
-        if path:
-            return config.load(path).as_dict()
-        else:
-            return {}
+    async def parse_response(request_params):
+        data = data_router.extract(request_params)
+        try:
+            return response.json(await data_router.parse(data), status=200)
+        except InvalidProjectError as e:
+            return response.json({"error": "{}".format(e)}, status=404)
+        except Exception as e:
+            logger.exception(e)
+            return response.json({"error": "{}".format(e)}, status=500)
 
-    @staticmethod
-    def _configure_logging(loglevel, logfile):
-        if logfile is None:
-            utils.configure_colored_logging(loglevel)
-        else:
-            logging.basicConfig(filename=logfile, level=loglevel)
-        logging.captureWarnings(True)
+    @app.get("/parse")
+    @requires_auth(app, token)
+    async def parse(request):
+        request_params = request.raw_args
 
-    @app.route("/", methods=["GET", "OPTIONS"])
-    @check_cors
-    def hello(self, request):
-        """Main Rasa route to check if the server is online"""
-        return "hello from Rasa NLU: " + rasa.__version__
-
-    @app.route("/parse", methods=["GET", "POST", "OPTIONS"])
-    @requires_auth
-    @check_cors
-    @inlineCallbacks
-    def parse(self, request):
-        request.setHeader("Content-Type", "application/json")
-        if request.method.decode("utf-8", "strict") == "GET":
-            request_params = decode_parameters(request)
-        else:
-            request_params = simplejson.loads(
-                request.content.read().decode("utf-8", "strict")
-            )
+        if "query" in request_params:
+            request_params["q"] = request_params.pop("query")
+        if "q" not in request_params:
+            request_params["q"] = ""
+        return await parse_response(request_params)
+
+    @app.post("/parse")
```

Comment:
can't we use the same function for both, `POST` and `GET` ? The functions seem the same to me

Reply:
Will revisit this once we actually merge the two servers.